### ISSUE TITLE
Bug: fail to load cmdline_cfg[":"] 

### ISSUE THREAD

**Issue by y1rn** [ibhagwan/fzf-lua#1164]
### RTFM Checklist

- [X] I have searched exisiting issues / discussions
- [X] I have read the Wiki including the Advanced section
- [X] I have read `man fzf` / I am well versed in shell fzf

### Operating system

Linux

### Shell

zsh

### Neovim version (`nvim --version`)

NVIM v0.10.0-dev-3045+gefaf37a2b

### Fzf version (`fzf --version`)

0.50.0 (d8bfb671)

### Output of `:lua print(os.getenv('FZF_DEFAULT_OPTS'))`

--layout=reverse --bind "change:top,ctrl-b:page-up,ctrl-d:page-down"

### Is the problem reproducible with `mini.sh`?

- [ ] My issue is reproducible with `mini.sh`
- [ ] My issue **IS NOT** reproducible with `mini.sh`
- [ ] I have not tested with `mini.sh` (not relevant, requires LSP, Windows, etc)

### Fzf-lua configuration

```lua
require('fzf-lua').setup({
})
```

### Describe the bug / steps to reproduce

show below error message:
```
Failed to source `/root/.local/share/nvim/lazy/fzf-lua/plugin/fzf-lua.lua`

vim/_editor.lua:0: /home/share/.vim/init.lua..nvim_exec2() called at /home/share/.vim/init.lua:0../root/.local/share/nvim/lazy/fzf-lua/plugin/fzf-lua.lua: Vim(source):E5113: Error while calling lua chunk: ...t/.local/share/nvim/lazy/fzf-lua/lua/fzf-lua/cmp_src.lua:69: attempt to index field ':' (a nil value)
stack traceback:
	...t/.local/share/nvim/lazy/fzf-lua/lua/fzf-lua/cmp_src.lua:69: in function '_register_cmdline'
	/root/.local/share/nvim/lazy/fzf-lua/plugin/fzf-lua.lua:32: in main chunk
	[C]: in function 'nvim_exec2'
	vim/_editor.lua: in function 'cmd'
	...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/loader.lua:485: in function <...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/loader.lua:484>
	[C]: in function 'xpcall'
	.../.local/share/nvim/lazy/lazy.nvim/lua/lazy/core/util.lua:113: in function 'try'
	...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/loader.lua:484: in function 'source'
	...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/loader.lua:443: in function 'source_runtime'
	...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/loader.lua:411: in function 'packadd'
	...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/loader.lua:346: in function '_load'
	...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/loader.lua:191: in function 'load'
	...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/loader.lua:123: in function 'startup'
	/root/.local/share/nvim/lazy/lazy.nvim/lua/lazy/init.lua:112: in function 'setup'
	/home/share/.vim/init.lua:70: in main chunk

# stacktrace:
  - vim/_editor.lua:0 _in_ **cmd**
  - /home/share/.vim/init.lua:70

```

looks `cmdline_cfg[":"]` is nil , but `lua require("cmp.config").cmdline[":"]` has values after neovim startup

here is my lazy configuration:
``` lua
{
  'hrsh7th/nvim-cmp',
  'hrsh7th/cmp-buffer',
  --  'hrsh7th/cmp-omni',
  'hrsh7th/cmp-path',
  'hrsh7th/cmp-cmdline',
  'hrsh7th/cmp-nvim-lsp',
  'hrsh7th/cmp-nvim-lua',
  'hrsh7th/cmp-nvim-lsp-signature-help',


  'hrsh7th/cmp-vsnip',
  'onsails/lspkind-nvim',
  'williamboman/mason.nvim',
  'williamboman/mason-lspconfig.nvim',
  'jayp0521/mason-nvim-dap.nvim',
  'hrsh7th/vim-vsnip',

  "ibhagwan/fzf-lua",
}
```

**Comment by kapral18** [comment]
seems like similar issue for me as well

sh -c "$(curl -s https://raw.githubusercontent.com/ibhagwan/fzf-lua/main/scripts/mini.sh)"

**Comment by idanarye** [comment]
This seems to have been introduced in eed06a7a9a28aeb47659b609999813d1bedc1a00, so I figure maybe it's expecting some new `cmp`-related configuration (and does not fail gracefully when that setting cannot be found), but the new docs don't mention anything of the sort...

**Comment by ibhagwan** [comment]
Sorry about that everyone, it was late night and I rushed to to merge a feature branch due to it containing a fix for another issue ( #1161).

Should not happen post https://github.com/ibhagwan/fzf-lua/commit/bb421160c790e00483789651d5ce7f3431103227.

**Comment by ibhagwan** [comment]
> looks cmdline_cfg[":"] is nil , but lua require("cmp.config").cmdline[":"] has values after neovim startup

Btw, regarding this, I would assume this happens when fzf-lua is loaded before-cmp, it’s not a big deal it just means that you’ll lose the docs for the pickers/options.

Overall, this will need some improvements, I’m not totally happy with the solution I came up with to have nvim-cmp display docs for pickers/options, still waiting on the maintainer of nvim-cmp for guidance:
https://github.com/hrsh7th/nvim-cmp/discussions/1885

When this works it adds auto documentation to the `:FzfLua` auto complete:
![IMG_2062](https://github.com/ibhagwan/fzf-lua/assets/59988195/8ad71aad-6bea-465b-b718-00b5d1dd9fdc)
